byway/src/main/webapp/track-interests.js

```diff
@@ -83,22 +84,22 @@ function createButtonForPlace(place) {
   button.innerText = place;
   button.addEventListener('click', () => {
       updateStatus(place, button);
-      fetchPlaces(tripId, interestsChosen);
+      fetchPlaces(tripKey, interestsChosen);
   });
   button.className = "interestBtn";
   return button;
 }
 
 /**
  * Gets all potential interests from server to load or sets the user's
- * specific interests for their trip with tripId. userInterests
+ * specific interests for their trip with tripKey. userInterests
  * may be any type that is convertible to an array via Array.from. It will
  * be sent to the server as JSON in the post body.
- * @param {String} tripId value for tripId
+ * @param {String} tripKey value for tripKey
  * @param {Array} [userInterests] interests selected by user
  */
-function fetchPlaces(tripId, /* optional */ userInterests) {
-  const url = '/api/places?' + new URLSearchParams({tripId}).toString();
+function fetchPlaces(tripKey, /* optional */ userInterests) {
+  const url = '/api/places?' + new URLSearchParams({tripKey}).toString();
```

Comment:
could you use the configureTripKeyForNextPage function for this line too?

Reply:
Good point, updated the function